code_generation/producers/scalefactors.py

```diff
@@ -57,3 +87,101 @@
         ],
     },
 )
+MuonIDIso_SF_old = ProducerGroup(
+    name="MuonIDIso_SF_old",
+    call=None,
+    input=None,
+    output=None,
+    scopes=["mt", "em", "mm"],
+    subproducers={
+        "mt": [
+            Muon_1_ID_SF_old,
+            Muon_1_Iso_SF_old,
+        ],
+        "em": [
+            Muon_2_ID_SF_old,
+            Muon_2_Iso_SF_old,
+        ],
+        "mm": [
+            Muon_1_ID_SF_old,
+            Muon_1_Iso_SF_old,
+            Muon_2_ID_SF_old,
+            Muon_2_Iso_SF_old,
+        ],
+    },
+)
+
+############################
+# Tau ID/ISO SF
+# The readout is done via correctionlib
+############################
+Tau_1_VsJetTauID_SF = ExtendedVectorProducer(
+    name="VsJetTauID_SF",
+    call='scalefactor::tau::id({df}, {input}, {vec_open}{tau_dms}{vec_close}, "{vsjet_tau_id_WP}", "{tau_sf_variation}", "{tau_sf_dependence}", {output}, "{tau_sf_file}", "DeepTau2017v2p1VSjet")',
+    input=[q.pt_1, q.eta_1, q.decaymode_1, q.gen_match_1],
+    output="tau_1_vsjet_sf_outputname",
+    scope=["mt"],
+    vec_config="vsjet_tau_id",
+)
+Tau_1_VsEleTauID_SF = ExtendedVectorProducer(
+    name="VsEleTauID_SF",
+    call='scalefactor::tau::id({df}, {input}, {vec_open}{tau_dms}{vec_close}, "{vsele_tau_id_WP}", "{tau_sf_variation}", "{tau_sf_dependence}", {output}, "{tau_sf_file}", "DeepTau2017v2p1VSe")',
+    input=[q.pt_1, q.eta_1, q.decaymode_1, q.gen_match_1],
+    output="tau_1_vsele_sf_outputname",
+    scope=["mt"],
+    vec_config="vsele_tau_id",
+)
+Tau_1_VsMuTauID_SF = ExtendedVectorProducer(
+    name="VsMuTauID_SF",
+    call='scalefactor::tau::id({df}, {input}, {vec_open}{tau_dms}{vec_close}, "{vsmu_tau_id_WP}", "{tau_sf_variation}", "{tau_sf_dependence}", {output}, "{tau_sf_file}", "DeepTau2017v2p1VSmu")',
+    input=[q.pt_1, q.eta_1, q.decaymode_1, q.gen_match_1],
+    output="tau_1_vsmu_sf_outputname",
+    scope=["mt"],
+    vec_config="vsmu_tau_id",
+)
+Tau_2_VsJetTauID_SF = ExtendedVectorProducer(
+    name="VsJetTauID_SF",
+    call='scalefactor::tau::id({df}, {input}, {vec_open}{tau_dms}{vec_close}, "{vsjet_tau_id_WP}", "{tau_sf_variation}", "{tau_sf_dependence}", {output}, "{tau_sf_file}", "DeepTau2017v2p1VSjet")',
+    input=[q.pt_2, q.eta_2, q.decaymode_2, q.gen_match_2],
+    output="tau_2_vsjet_sf_outputname",
+    scope=["mt"],
```

Comment:
`scope = ["et", "mt", "tt"]`

Reply:
Done

Reply:
hmm, then we have to take a closer look into the `ExtendedVectorProducer` for this to work properly. I suggest keeping it as you had it in the beginning and fixing it in a seperate PR. Could you open an issue for this ?